spec/models/adviser/sign_up_spec.rb

```diff
@@ -38,6 +43,24 @@
 
       it { is_expected.not_to be_available }
     end
+
+    context 'when the Get into Teaching API is unavailable' do
+      before do
+        error = GetIntoTeachingApiClient::ApiError.new(code: 500)
+        allow_any_instance_of(GetIntoTeachingApiClient::TeacherTrainingAdviserApi)
```

Comment:
is it possible to use
```
let(:ttapi) { instance_double(GetIntoTeachingApiClient::TeacherTrainingAdviserApi ...) }
allow(GetIntoTeachingApiClient::TeacherTrainingAdviserApi).to receive(:new).and_return(ttapi)
```

we've managed to keep our `allow_any_instance_of` count very low so far!